<?xml version="1.0"?>
<launch>

    <!--    Position and angle of the limb  -->
    <arg name="x" default="0.0"/>
    <arg name="y" default="0.0"/>
    <arg name="yaw" default="0.0"/>

    <!--    Number of the limb for creating an unique ID    -->
    <arg name="limb_number" default="0"/>

    <arg name="load_control" default="true"/>

    <!--    Type of base joint  -->
    <arg name="base_joint_type" default="revolute"/>

    <group ns="limb$(arg limb_number)">
        <!-- Load the URDF into the ROS Parameter Server -->
        <param name="robot_description"  command="$(find xacro)/xacro '$(find robot_description)/urdf/limb_base_$(arg base_joint_type).xacro' x:=$(arg x) y:=$(arg y) yaw:=$(arg yaw)" />

        <!-- Run a python script to the send a service call to gazebo_ros to spawn a URDF robot -->
        <node name="urdf_spawner" pkg="gazebo_ros" type="spawn_model" respawn="false" output="screen"
                args="-urdf -model limb$(arg limb_number) -param robot_description"/>

        <!-- Load joint controller configurations from YAML file to parameter server -->
        <rosparam file="$(find robot_description)/config/$(arg base_joint_type)_joint_config.yaml" command="load" if="$(eval arg('load_control'))"/>

        <param name="is_extendible" type="bool" value="true" unless="$(eval arg('load_control'))"/>

    </group>

</launch>
